projects/HotFire/config/UOStateMachine.cpp

```diff
@@ -325,8 +325,7 @@ STATE_DEFINE(UOStateMachine, AbortBurn, UOSMData)
 
 void UOStateMachine::detectExternEvent(const sensorsData *data)
 {
-#if USE_INPUT
-	int eventNbr = data->inputEventNumber;
+	int eventNbr = data->eventNumber;
```

Comment:
Instead of doing the switch statement bellow, we could instead take the eventNbr, do some "if conditions" on it:
- if it is -1, then return; (don't do anything because -1 is the default value)
- If eventNbr is not in the range 0 ---- ST_MAX_STATES-1 (both included) then don't do anything and print an error message

Then, I think that we will be able to simply do InternalEvent(eventNbr); or something similar (we might need to cast something in the process, so be careful)

But if we do it this way, it will be more flexible and the code will be cleaner

Reply:
I take back what I said. We need to call the function because we need the transition map. However, maybe we could have a "list of function" that we could use to map the eventNbr to call the function call for the event. I don't know how that's possible in c++, but I'm sure there is a way

Reply:
I think a switch is more readable than a list. 

Also, I changed this part slightly in https://github.com/uorocketry/rocket-code-2020/tree/two-way-radio (moved the abort events all into one).